src/components/tl/ucp/scatter/scatter_knomial.c

```diff
@@ -62,13 +63,7 @@ void ucc_tl_ucp_scatter_knomial_progress(ucc_coll_task_t *coll_task)
     while (!ucc_knomial_pattern_loop_done(p)) {
         step_radix  = ucc_kn_compute_step_radix(p);
         block_count = ucc_sra_kn_compute_block_count(count, rank, p);
-        sbuf        = (rank == root)
-                           ? args->src.info.buffer : args->dst.info.buffer;
-        rbuf        = args->dst.info.buffer;
```

Comment:
where are you setting rbuf instead?

Reply:
can you elaborate pls, i don't understand the question

Reply:
nevermind, i see it is already set in the begining of progress and this is redundant.